docs/component_principles.md

```diff
@@ -0,0 +1,53 @@
+# Component Principles
+
+What makes a good component?
+
+A component is a package comprised of template, style, behaviour and documentation.
+
+## A good component:
+
+* is accessible
+* is documented
+* [follows convention](component_conventions.md)
+* is isolated
+* is tested
+* can be translated
```

Comment:
We've not talked about this much since starting this component work. Is there something we can link to here to give a bit more detail about how translations should be handled in applications?